flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableResultImpl.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api.internal;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.core.execution.JobClient;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.ResultKind;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Implementation for {@link TableResult}.
+ */
+@Internal
+public class TableResultImpl implements TableResult {
+	public static final TableResult TABLE_RESULT_OK = TableResultImpl.builder()
+			.resultKind(ResultKind.SUCCESS)
+			.tableSchema(TableSchema.builder().field("result", DataTypes.STRING()).build())
+			.data(Collections.singletonList(Row.of("OK")))
+			.build();
+
+	private final JobClient jobClient;
+	private final TableSchema tableSchema;
+	private final ResultKind resultKind;
+	private final Iterator<Row> data;
+
+	private TableResultImpl(
+			@Nullable JobClient jobClient,
+			TableSchema tableSchema,
+			ResultKind resultKind,
+			Iterator<Row> data) {
+		this.jobClient = jobClient;
+		this.tableSchema = Preconditions.checkNotNull(tableSchema, "tableSchema should not be null");
+		this.resultKind = Preconditions.checkNotNull(resultKind, "resultKind should not be null");
+		this.data = Preconditions.checkNotNull(data, "data should not be null");
+	}
+
+	@Override
+	public Optional<JobClient> getJobClient() {
+		return Optional.ofNullable(jobClient);
+	}
+
+	@Override
+	public TableSchema getTableSchema() {
+		return tableSchema;
+	}
+
+	@Override
+	public ResultKind getResultKind() {
+		return resultKind;
+	}
+
+	@Override
+	public Iterator<Row> collect() {
+		return data;
+	}
+
+	@Override
+	public void print() {
```

Comment:
how about reusing some logic with tableau format in SQL CLI, which you can find in `CliTableauResultView`